### PR TITLE
Primary Helm chart for fission

### PR DESCRIPTION
**Pull request by ssudake21**
This is attempt to solve #62. Created basic Helm Chart for Fission referring current installation process. etcd-operator can be used as charts requirement here.

Chart REAME:
https://github.com/ssudake21/fission/tree/helm_chart/charts/fission

Demo:
https://asciinema.org/a/ecmn9l5iypzdd6i3m9vu6rph4

### CODE REVIEW

**Comment by ssudake21** [comment]
Will reopen later.

**Comment by soamvasani** [comment]
Wow, this looks pretty good!  Thanks for taking the time to make this chart.  

Was something wrong with this pull request?

**Comment by ssudake21** [comment]
Seems like some existing failures are there in master. I have rebased my changes with master.

**Comment by soamvasani** [comment]
Hey, thanks for this contribution! This is great. I'm going to take a few days to review since I'm very new to Helm :)

**Comment by ssudake21** [comment]
Sure. Take your time.

File: charts/fission/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+FISSION_URL is used by the fission CLI to find the server.
+FISSION_URL should be prefixed with a http://. (FISSION_ROUTER is only needed for the examples below to work.)
+
+{{- if contains "NodePort" .Values.serviceType }}
+  $ export FISSION_URL=http://$(minikube ip):{{ .Values.controllerPort }}
+  $ export FISSION_ROUTER=$(minikube ip):{{ .Values.routerPort }}
+
+{{- else if contains "LoadBalancer" .Values.serviceType }}
+  NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+        You can watch the status of by running 'kubectl --namespace fission get -w svc' and 'kubectl get svc -w router'
```
Comment (soamvasani) [inline charts/fission/templates/NOTES.txt]: The second command would need a `--namespace fission` right?

File: charts/fission/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  $ export FISSION_URL=http://$(kubectl --namespace fission get svc controller -o=jsonpath='{..ip}')
+  $ export FISSION_ROUTER=$(kubectl --namespace fission get svc router -o=jsonpath='{..ip}')
+
+{{- end }}
+
+2. Install the client CLI.
```
Comment (soamvasani) [inline charts/fission/templates/NOTES.txt]: I would make this step 1, and the env var setup step 2.

**Comment by soamvasani** [comment]
Hey @ssudake21 , just a couple of minor comments above.  Can you also describe the testing done?

**Comment by ssudake21** [comment]
Hi @soamvasani ,

I have tested on GKE thoroughly.  I have tested  with minicube when I started developing chart I will test with minicube once again (having some setup issues).
I will need help with AWS as I don't have account as of now.

**Comment by soamvasani** [comment]
@ssudake21 thanks, sounds great.  Let me know when you finish testing with minikube, I'll merge it then.

**Comment by ssudake21** [comment]
Verified with minikube.

https://gist.github.com/ssudake21/cd9dc79296f2a8099d10342842d4f649

**Comment by soamvasani** [comment]
🎉 Awesome, thanks @ssudake21 !

**Pull request merged**